dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/sharing/Sharing.java

```diff
@@ -211,4 +220,37 @@ public Sharing copy()
             .users( new HashMap<>( users ) )
             .userGroups( new HashMap<>( userGroups ) ).build();
     }
+
+    public Sharing withAccess( UnaryOperator<String> accessTransformation )
```

Comment:
Maybe move out of the Sharing domain class to a SharingUtils class, to keep the domain class cleaner.

Reply:
I'd consider this an anti-pattern as it forces static helper methods that are only dependent on a single parameter which indicates that this rather should be a instance method of the single parameter type. 
I can understand that one might want to create static method anyhow in case the operation is very specific but I cannot see how this applies here. Moving this to a static helper also increases the risk of duplication as people don't see the method already exist when they are unaware of the particular helper class or don't check there first. WDYT?